File: js/two-sum/two-sum.js
function twoSum(nums, target){
    let numToIndexMap={}
    for(let i=0 ; i<nums.length ; i++){
        let diff=target-nums[i]
        if (diff in numToIndexMap){
            return [i, numToIndexMap[diff]]
        }
        numToIndexMap[nums[i]]=i
    }
    return []
}

console.log("answer to the first example is:",twoSum([3, 2, 4], 6))

File: go/two-sum/two-sum.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) []int {
    numToIndexMap := make(map[int]int)

    for i, num := range nums {
        
        diff := target - num

        if idx, found := numToIndexMap[diff]; found {
            return []int{i, idx}
        }

        numToIndexMap[num] = i
    }
	return nil
}

func main(){
	twoSum([]int{3, 2, 4}, 6)
	fmt.Println("answer to the first example is:", twoSum([]int{3, 2, 4}, 6))
}